reporter.cpp

```diff
@@ -51,20 +51,31 @@ void segfault_handler(int signal_number) { siglongjmp(env, 1); }
 // the fuzzer file will be linked to multiple targets
 // for each target, the table should be dumped once
 thread_local unsigned int dump_counter = 0;
+char *dump_file_name = "tmp_dump.json";
```

Comment:
You can do:
```cpp
class DumpFileNameSetter {
   static char* filename
   DumpFileNameSetter() {
       filename = getenv(...);
   }
}
static DumpFileNameSetter  dump_file_name_setter();
```
So that this piece of code is executed whenever your reporter is loaded, and you don't need to instrument it.

Reply:
If we use env variable to set dump file name, we can just use something like

```cpp
static char* FILENAME = "";
__attribute__((constructor)) static void get_dump_filename() {
 FILENAME = std::getenv(...);
}
```